### DISCUSSION TITLE
YOLOv8 Transfer Learning

### DISCUSSION THREAD

**Discussion by hgollnick** [ultralytics/ultralytics#1431] [Q&A]
Is there a way to apply Transfer Learning on YOLOv8? if that is the case, where could I find more documentation on the subject?

**Comment by KatarinaPopikova** [comment]
I think, 
https://docs.ultralytics.com/modes/train/
in **Usage Examples** section

**Python**
```
model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)
# Train the model
model.train(data='coco128.yaml', epochs=100, imgsz=640)
```

**CLI**
```
# Start training from a pretrained *.pt model
yolo detect train data=coco128.yaml model=yolov8n.pt epochs=100 imgsz=640
```

**Reply by hgollnick** [reply]
I don't know if there is any more steps on the process, since when I try training a pretrained *.pt model I can't maintaing the interfacing present on the pretrained model + the one that i've trained on my custom dataset...

e.g. 

yolov8n.pt can detect dogs
mymodel.pt can detect mugs

yolov8n.pt + mymodel.pt can only detect mugs

I apologize if the explanation is still a bit confusing...

**Reply by glenn-jocher** [reply]
@hgollnick no worries, your question is clear. When you train a pretrained model on a new custom dataset, it's important to ensure that the new dataset includes all the classes you want the model to recognize post-training. If you train only on mugs, the model will specialize in mugs and likely forget about dogs.

To maintain the ability to detect both dogs and mugs, you should combine the datasets (i.e., have images of both dogs and mugs with appropriate labels) before training. This way, the model learns to detect both classes during the transfer learning process.

Remember to update your dataset configuration file to include all the classes you want to detect. Happy training! 🚀

**Comment by Y-T-G** [comment]
Transfer learning doesn't mean the model wouldn't forget what it learnt before. Unless you're providing all the classes the model has to detect, it will eventually forget those classes since the goal during training is to maximize performance on the current dataset. Since your current dataset doesn't have those classes, the weights won't be updated in a way to preserve the performance on those classes.
